lib/Dialect/TritonNvidiaGPU/Transforms/MaterializeLoadStore.cpp

```diff
@@ -202,6 +219,90 @@ void MaterializeLoadStorePass::materializeStoreTilePtr(
   store->erase();
 }
 
+SmallVector<Value>
+MaterializeLoadStorePass::immutableMbarrierAlloc(Operation *entryOp,
+                                                 unsigned loadNum) {
+  SmallVector<Value> barrierLists;
+  OpBuilder builder(entryOp);
+  Location loc = entryOp->getLoc();
+  builder.setInsertionPoint(entryOp);
+  for (unsigned i = 0; i < loadNum; ++i) {
+    auto mBarrierTy =
+        mlir::triton::PointerType::get(builder.getIntegerType(64), 3);
+    Value mBarrier = builder.create<ttng::AllocMBarrierOp>(loc, mBarrierTy, 1);
+    barrierLists.push_back(mBarrier);
+  }
+  return barrierLists;
```

Comment:
barrierLists is better to be the private member of this class, to avoid unnecessary copies

Reply:
Previously, considering the RVO optimazation([https://www.geeksforgeeks.org/copy-elision-in-cpp/]()),  I use an explicite copy to get a better function encapsulation（this func can be stateless）.but now, I can't ensure this RVO optimaztion is enable, so I have modified it as a private member